life_expectancy/cleaning.py

```diff
@@ -0,0 +1,77 @@
+import pandas as pd
+import argparse
+from pathlib import Path
+
+def load_data() -> pd.DataFrame:
+    """ reads life expectancy data into a DataFrame"""
+    path = Path().cwd() / 'life_expectancy' / 'data' / 'eu_life_expectancy_raw.tsv'
+    return pd.read_csv(path, sep = '\t')
+
+def split_first_column(df: pd.DataFrame) -> pd.DataFrame:
+    """ Splits four variables in the first column into their own columns """
+    df[['unit', 'sex', 'age', 'region']] = df[df.columns[0]].str.split(
+                                           ',', expand=True)
+    df = df.drop(columns = df.columns[0])
+    return df
+
+def remove_spaces_from_column_names(df: pd.DataFrame) -> pd.DataFrame:
+    """ Removes extra space from column names."""   
+    df = df.rename(str.strip, axis = 'columns')
+    return df
+    
+def melt_df(df: pd.DataFrame) -> pd.DataFrame:
+    """ Melts DataFrame. """
+    years = df.columns[:-4]
+    return pd.melt(
+        df, 
+        id_vars = ['unit', 'sex', 'age', 'region'], 
+        value_vars = years, var_name = 'year', 
+        value_name = 'value')
+
+def extract_flag(df: pd.DataFrame) -> pd.DataFrame:
+    """Extract flag (b, e, p or their combination)
+    from the value column."""
+    df[['value','flag']] = df['value'].str.split(' ', expand=True)
+    return df
+
+def correct_data_types(df: pd.DataFrame) -> pd.DataFrame:
+    for col in ['unit', 'sex', 'age', 'region','flag']:
+        df[col] = df[col].astype('string')
+    df['year'] = df['year'].astype('int')
+    df['value'] = pd.to_numeric(
+        df['value'], errors = 'coerce')
+    return df
+
+def export_region_data(df: pd.DataFrame, region: str) -> None:
```

Comment:
```suggestion
def filter_region_data(df: pd.DataFrame, region: str) -> pd.DataFrame:
```
The function is returning a dataframe but type hint indicates None.

Also, in this function we are not exporting data but only filtering or slicing the dataframe so we could be more explicit in the function name. Smth like filter_region_data or slice_region_dataframe

Reply:
I changed the function name to filter_region_data and corrected the type hint.